docs/source/developer_guide/dependencies.rst

```diff
@@ -93,3 +93,19 @@ the following files need to be updated:
 
 *  `pyproject.toml <https://github.com/sktime/sktime/blob/main/pyproject.toml>`__,
    adding the dependency or version bounds in the ``dev`` dependency set.
+
+Maintenance of bounds for soft dependencies
```

Comment:
can we merge this with section "adding a soft dependency"? Maybe change name to "adding and maintaining", and include a paragraph that explains the dependency sets.

Reply:
addressed.